packages/core/http/core-http-server/index.ts

```diff
@@ -128,3 +128,14 @@ export type {
   HttpServiceSetup,
   HttpServiceStart,
 } from './src/http_contract';
+
+export type {
+  AddVersionOpts,
+  RequestValidation,
+  ResponseValidation,
+  Version,
+  VersionedRoute,
+  VersionedRouteConfig,
+  VersionedRouteRegistrar,
+  VersionedRouter,
+} from './src/versioning';
```

Comment:
NIT: I wonder if we shouldn't add a more explicit prefix to some of those types, just to make it more understandable that they're bound to the versioned router. Mostly thinking about:
- `Version`
- `RequestValidation`
- `ResponseValidation`

Maybe (I know, those are long names...):
- `RouteVersion` (or `ApiVersion`?)
- `VersionedRouteRequestValidation`
- `VersionedRouteResponseValidation`